libnmstate/prettystate.py

```diff
@@ -71,35 +69,42 @@ def json(self):
         return json.dumps(self.state, indent=4, separators=(",", ": "))
 
 
-def represent_ordereddict(dumper, data):
+def represent_dict(dumper, data):
+    """
+    Represent dictionary with sorted order and honoring the PRIORITY_LIST.
     """
-    Represent OrderedDict as regular dictionary
-
-    Source: https://stackoverflow.com/questions/16782112/can-pyyaml-dump-dict-items-in-non-alphabetical-order
-    """  # noqa: E501
     value = []
+    normal_items = {}
+    list_items = {}
+    dict_items = {}
 
     for item_key, item_value in data.items():
-        node_key = dumper.represent_data(item_key)
-        node_value = dumper.represent_data(item_value)
+        if isinstance(item_value, list):
+            list_items[item_key] = item_value
+        elif isinstance(item_value, dict):
```

Comment:
This should probably use abstract types like: https://docs.python.org/2/library/collections.html#collections.Mapping